File: src/283.move-zeroes.py
#
# @lc app=leetcode id=283 lang=python3
#
# [283] Move Zeroes
#
from typing import List
# @lc code=start
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        nums.sort(key=lambda x: x == 0)

# @lc code=end

File: Arrays 101/src/283.move-zeroes.js
/*
 * @lc app=leetcode id=283 lang=javascript
 *
 * [283] Move Zeroes
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {

  let cursor = 0;
  let zeroCount = 0;
  
  for (var i = 0; i < nums.length; i++) {
    if (nums[i] === 0) {
      zeroCount++;
    } else {
      nums[cursor++] = nums[i];
    }
  }

  for (var i = 0; i < zeroCount; i++) {
    nums[cursor++] = 0;
  }

};
// @lc code=end

